src/AggregateFunctions/AggregateFunctionMinMaxAny.h

```diff
@@ -542,25 +555,42 @@ struct SingleValueDataString //-V730
     {
         Int32 value_size = static_cast<Int32>(value.size);
 
+        if (value_size <= 0)
+        {
+            size = 1;
+            small_data[0] = '\0';
+            using_internal = true;
+            return;
+        }
+
         if (value_size <= MAX_SMALL_STRING_SIZE)
         {
             /// Don't free large_data here.
             size = value_size;
-
-            if (size > 0)
-                memcpy(small_data, value.data, size);
+            memcpy(small_data, value.data, size);
+            using_internal = true;
         }
         else
         {
-            if (capacity < value_size)
+            Int32 size_with_possible_null = value_size + 1;
+            if (capacity < size_with_possible_null)
             {
                 /// Don't free large_data here.
-                capacity = static_cast<Int32>(roundUpToPowerOfTwoOrZero(value_size));
+                capacity = static_cast<Int32>(roundUpToPowerOfTwoOrZero(size_with_possible_null));
                 large_data = arena->alloc(capacity);
             }
 
             size = value_size;
             memcpy(large_data, value.data, size);
+            using_internal = false;
+        }
+
+        char * data = getData();
+        if (data[size - 1] != '\0')
```

Comment:
Uhmm, yeah. That approach looks much better and only adds (less) complexity to the write operations. I'll look into it.

Reply:
I've tried this approach and, although it's much simpler, it still has the issue of the original NULL byte in the string. I'm pushing it for now for comments, but it might not be the final approach.